src/codecs/webp/decoder.ts

```diff
@@ -1,10 +1,43 @@
 import { blobToArrayBuffer } from '../../lib/util';
-import DecoderWorker  from './Decoder.worker';
+import { DecoderWorker } from '../codec-worker';
 
 export const name = 'WASM WebP Decoder';
+
+const POOL_SIZE = 4;
```

Comment:
I thought we were avoiding pooling in favour of two workers, one for the left and one for the right?

For the decode of the input image, we can just use the left worker.

In terms of workers, the flow is like this:

1. Label: `inputChange`.
1. Decode image if not natively supported format.
1. For each of left & right:
    1. Label: `encoderSettingsChanged`.
    1. If not "original image", encode image.
    1. Decode image if not natively supported format.

When the user selects a file, goto `inputChange`. When the user changes settings on the left, goto `encoderSettingsChanged` for the left. When the user changes settings on the right, goto `encoderSettingsChanged` for the right.

Because of this, the maximum number of decodes we need concurrently is 2. We can reuse one of the left/right workers for the initial encode, because once the input changes, whatever the left/right workers are currently doing becomes redundant.

Reply:
Not sure if I'm following correctly, but we can't reuse a worker for multiple codecs.

I'm going to take a stab at replacing the decoder pool here with an instantiation-based approach like I set up for encoders.